### DISCUSSION TITLE
Changing the row order on motif enrichment heat maps?

### DISCUSSION THREAD

**Discussion by edt711** [GreenleafLab/ArchR#574] [Feature Requests]
Hi ArchR team,

Thank you for putting together such an awesome program!! When making heat maps of motif enrichments, is there any way to manually set/change the order of the rows (which correspond to the clusters/cell types)? I know that you can set `binaryClusterRows = FALSE` in `plotEnrichHeatmap`, but then it just defaults to an alphabetical order. Is there a way to specify the order in which the clusters appear on the heat map, either through an ArchR command or via ComplexHeatmap?

Thanks,
Eric

**Comment by rcorces** [comment]
To the best of my knowledge this is not yet possible but is a good suggestion. I've added the enhancement label and we will look into it.

**Comment by edt711** [comment]
Awesome, thank you so much!

**Comment by cristanchoa** [comment]
Hi, 

Thanks again for the great package. I have been looking through the archives, but I can't tell if this enhancement for controlling the row and column order is available in the various heatmaps yet, or if you have any suggestions on how export an appropriate matrix to enter into complexheatmap.

Thank you, 
Ana

**Reply by rcorces** [reply]
This can be controlled via editing the `ComplexHeatmap` object.
Original heatmap - 
![image](https://user-images.githubusercontent.com/10343046/182613232-f1f3ccc5-8c2f-45e3-afa2-9e7df36680da.png)

```
> heatmapEM@row_order
 [1]  1  2  3  4  5  6  7  8  9 10 11
> temp <- heatmapEM@row_order
> heatmapEM@row_order <- sample(temp, size = length(temp), replace = FALSE)
> heatmapEM@row_order
 [1]  3  7  5  8  6  1  2  4 10 11  9
> heatmapEM
```

New heatmap:
![image](https://user-images.githubusercontent.com/10343046/182613380-9fa93562-cad7-4a0f-87b6-6c0b14d6c780.png)
